src/api/auth.ts

```diff
@@ -134,3 +134,234 @@ function anonymousUser(): any {
 function authenticatedUser(name: string, groups: string[] = []): any {
   return { name: name, groups: groups.concat(['&all', '$anonymous']), real_groups: groups };
 }
+
+export function newUser(req: express.Request, res: express.Response): express.Response {
+  let username = req.body.username;
+  let fullname = req.body.fullname;
+  let email = req.body.email;
+  let password = req.body.password;
+  let auth = getAuth();
+  let config = getConfig();
+  let hash = generateHash(password, config.secret);
+  let user = {
+    'name': username,
```

Comment:
there's no need to use single quotes here in object notation, so: `name: username` is just fine.